lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1249,8 +1249,7 @@ private function executeDeletions(ClassMetadata $class): void
      */
     private function getCommitOrder(): array
     {
-        $entityChangeSet = array_merge($this->entityInsertions, $this->entityUpdates, $this->entityDeletions);
-        $calc            = $this->getCommitOrderCalculator();
+        $calc = $this->getCommitOrderCalculator();
```

Comment:
Covered by

`\Doctrine\Performance\Hydration\SingleTableInheritanceInsertPerformanceBench`

and 

`\Doctrine\Performance\Hydration\SingleTableInheritanceHydrationPerformanceBench`